src/lib/party.ts

```diff
@@ -819,27 +703,100 @@ export class Party {
     }
 
     // if the fee payer is the same party as this one, we have to start the nonce predicate at one higher bc the fee payer already increases its nonce
-    let nonceIncrease = Circuit.if(
-      new Bool(options?.isSameAsFeePayer ?? false),
-      new UInt32(Field.one),
-      UInt32.zero
-    );
+    let nonceIncrement = options?.isSameAsFeePayer
+      ? new UInt32(Field.one)
+      : UInt32.zero;
     // now, we check how often this party already updated its nonce in this tx, and increase nonce from `getAccount` by that amount
     for (let party of Mina.currentTransaction.parties) {
       let shouldIncreaseNonce = party.publicKey
         .equals(publicKey)
         .and(party.body.incrementNonce);
-      nonceIncrease.add(new UInt32(shouldIncreaseNonce.toField()));
+      nonceIncrement.add(new UInt32(shouldIncreaseNonce.toField()));
     }
-    let nonce = account.nonce.add(nonceIncrease);
+    let nonce = account.nonce.add(nonceIncrement);
 
     body.accountPrecondition = nonce;
+    body.incrementNonce = Bool(true);
 
     let party = new Party(body) as Party & {
       body: { accountPrecondition: UInt32 };
     };
+    party.sign(signer);
     Mina.currentTransaction.nextPartyIndex++;
     Mina.currentTransaction.parties.push(party);
     return party;
   }
 }
+
+type FeePayer = Party & {
+  authorization: Exclude<LazyControl, { kind: 'proof'; value: string }>;
+} & {
+  body: { accountPrecondition: UInt32 };
+};
+
+type Parties = { feePayer: FeePayer; otherParties: Party[] };
+
+function addMissingSignatures(
```

Comment:
same here -- ideally we have a separate type for a "pending signatures Party" and a "finished Party"

Reply:
See ExternalTransition.t and ExternalTransition.Validated.t in the OCaml codebase for an example of this

Reply:
done, I made a type `PartiesValidated` where the signature can't be lazy https://github.com/o1-labs/snarkyjs/pull/123/commits/f10aca039cfda0630ffddbbc9ca3327f814a4d40